pulsar-client/src/main/java/org/apache/pulsar/client/api/ProducerConfiguration.java

```diff
@@ -423,6 +427,23 @@ public ProducerConfiguration setInitialSequenceId(long initialSequenceId) {
         return this;
     }
 
+    /**
+     * Set optional key/value metadata with this producer.
+     * @param key
+     * @param value
+     * @return
+     */
+    public ProducerConfiguration setMetadata(String key, String value) {
```

Comment:
Instead of calling it `metadata`, what about reusing the term `properties` like on the `Message` itself? like: https://github.com/apache/incubator-pulsar/blob/master/pulsar-client/src/main/java/org/apache/pulsar/client/api/MessageBuilder.java#L86

Reply:
Updated the configuration interface to use property methods like the message builder